### PR TITLE
Minor: Fixed regexpr_match docs

### LINKED ISSUE #13004: Minor: fix regexp_match function documentation
### Describe the bug

The current regexp_match documentation says that:
```
Returns a list of [regular expression](https://docs.rs/regex/latest/regex/#syntax) matches in a string.
```

In fact, it only returns the first match, not a list. The proposed [regexp_matches](https://github.com/apache/datafusion/issues/11946) function would return the list.

### To Reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonathanc-n**
## Which issue does this PR close?

Closes #13004 .

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Thank you @jonathanc-n 

I also took the liberty of regenerating the docs and merging this branch from main

**Comment by jonathanc-n** [comment]
@alamb Thank you, sorry forgot to regenerate.

**Comment by alamb** [comment]
Thanks again @jonathanc-n

File: datafusion/functions/src/regex/regexpmatch.rs [hunk 1]
```diff
@@ -119,7 +119,7 @@ fn get_regexp_match_doc() -> &'static Documentation {
     DOCUMENTATION.get_or_init(|| {
         Documentation::builder()
             .with_doc_section(DOC_SECTION_REGEX)
-            .with_description("Returns a list of [regular expression](https://docs.rs/regex/latest/regex/#syntax) matches in a string.")
+            .with_description("Returns the first [regular expression](https://docs.rs/regex/latest/regex/#syntax) matche in a string.")
```
Comment (Omega359) [inline datafusion/functions/src/regex/regexpmatch.rs:122]: matche ?

Comment (jonathanc-n) [reply datafusion/functions/src/regex/regexpmatch.rs:122]: Oh, i'll fix that rlly quick.

Comment (alamb) [reply datafusion/functions/src/regex/regexpmatch.rs:122]: fixed in https://github.com/apache/datafusion/pull/13014

**Pull request merged**